### PR TITLE
Add BlobServiceClient client to get blobServiceProperties

### PR DESCRIPTION
**Pull request by ndowmon**

### CODE REVIEW

File: src/steps/resource-manager/storage/__recordings__/storageAccountServiceClient-getBlobServiceProperties_1284191461/recording.har [hunk 1]
```diff
@@ -541,121 +541,145 @@
...
-          ],
-          "url": "https://management.azure.com/subscriptions/d3803fd6-2ba4-4286-80aa-f3d613ad59a7/resourceGroups/j1dev/providers/Microsoft.Storage/storageAccounts/ndowmon1j1dev/blobServices/default?api-version=2019-06-01"
+          "method": "POST",
+          "postData": {
+            "mimeType": "application/x-www-form-urlencoded",
+            "params": [],
+            "text": "[REDACTED]"
+          },
+          "queryString": [],
+          "url": "https://login.microsoftonline.com/992d7bbe-b367-459c-a10f-cf3fd16103ab/oauth2/v2.0/token"
```
Comment (ndowmon) [inline src/steps/resource-manager/storage/__recordings__/storageAccountServiceClient-getBlobServiceProperties_1284191461/recording.har:566]: @aiwilliams I noticed that the `authenticate` function you built, the `tokenAudience` is set to `management.azure.com` (https://github.com/JupiterOne/graph-azure/blob/master/src/azure/resource-manager/authenticate.ts#L65).

```typescript
  const response = await msRestNodeAuth.loginWithServicePrincipalSecretWithAuthResponse(
    config.clientId,
    config.clientSecret,
    config.directoryId,
    {
      tokenAudience: 'https://management.azure.com/',
    },
  );
```

I use the `ClientSecretCredential` class for authentication of this non-ARM client (thanks for your very helpful [comment](https://github.com/JupiterOne/graph-azure/blob/cis-benchmark-3.10/src/azure/graph/authenticate.ts#L14)!) which defaults to using `login.microsoftonline.com` as the audience... I suppose (I may not be understanding the "tokenAudience" fully).

Wondering your reasoning behind originally using `management.azure.com`, and if you recognize a foreseeable issue authenticating in this way.

File: src/steps/resource-manager/storage/__recordings__/storageAccountServiceClient-getBlobServiceProperties_1284191461/recording.har [hunk 2]
```diff
@@ -697,114 +716,80 @@
...
+            "mimeType": "application/xml",
+            "size": 815,
+            "text": "﻿<?xml version=\"1.0\" encoding=\"utf-8\"?><StorageServiceProperties><Logging><Version>1.0</Version><Read>true</Read><Write>true</Write><Delete>true</Delete><RetentionPolicy><Enabled>true</Enabled><Days>7</Days></RetentionPolicy></Logging><HourMetrics><Version>1.0</Version><Enabled>true</Enabled><IncludeAPIs>true</IncludeAPIs><RetentionPolicy><Enabled>true</Enabled><Days>7</Days></RetentionPolicy></HourMetrics><MinuteMetrics><Version>1.0</Version><Enabled>true</Enabled><IncludeAPIs>true</IncludeAPIs><RetentionPolicy><Enabled>true</Enabled><Days>7</Days></RetentionPolicy></MinuteMetrics><Cors /><DeleteRetentionPolicy><Enabled>true</Enabled><Days>7</Days><AllowPermanentDelete>false</AllowPermanentDelete></DeleteRetentionPolicy><StaticWebsite><Enabled>false</Enabled></StaticWebsite></StorageServiceProperties>"
```
Comment (ndowmon) [inline src/steps/resource-manager/storage/__recordings__/storageAccountServiceClient-getBlobServiceProperties_1284191461/recording.har:744]: FYI - this different client returns XML, but many of these properties that we need for CIS are not returned in the ARM-based client. You can clearly see how different the URL of this request is constructed, compared to what we see in the other client.

File: src/steps/resource-manager/storage/converters.ts [hunk 3]
```diff
@@ -95,6 +95,12 @@ export function createStorageAccountEntity(
           storageBlobServiceProperties?.deleteRetentionPolicy?.enabled,
         blobSoftDeleteRetentionDays:
           storageBlobServiceProperties?.deleteRetentionPolicy?.days,
+        blobAnalyticsLoggingReadEnabled:
+          storageBlobServiceProperties.blobAnalyticsLogging?.read,
+        blobAnalyticsLoggingWriteEnabled:
+          storageBlobServiceProperties.blobAnalyticsLogging?.write,
+        blobAnalyticsLoggingDeleteEnabled:
+          storageBlobServiceProperties.blobAnalyticsLogging?.deleteProperty,
```
Comment (ndowmon) [inline src/steps/resource-manager/storage/converters.ts:103]: Thoughts on these property names?

File: src/steps/resource-manager/storage/client.ts [hunk 4]
```diff
@@ -1,19 +1,46 @@
...
 import {
   Client,
   iterateAllResources,
 } from '../../../azure/resource-manager/client';
 import { resourceGroupName } from '../../../azure/utils';
+import { IntegrationConfig } from '../../../types';
+import { IntegrationLogger } from '@jupiterone/integration-sdk-core';
+import { ClientSecretCredential } from '@azure/identity';
+
+export function createStorageAccountServiceClient(options: {
```
Comment (ndowmon) [inline src/steps/resource-manager/storage/client.ts:21]: FYI - this client will soon have additional methods `getTableServiceProperties` and `getQueueServiceProperties`.

File: src/steps/resource-manager/storage/__recordings__/storageAccountServiceClient-getBlobServiceProperties_1284191461/recording.har (hunk 1 above)
Comment (aiwilliams) [reply src/steps/resource-manager/storage/__recordings__/storageAccountServiceClient-getBlobServiceProperties_1284191461/recording.har:72]: FTR... my recollection is that the original authentication code was written by someone else. When I began working on the MS365 integration, I discovered the `ClientSecretCredential` mechanisms and used them in that integration. It worked for my purposes and I did not have reasons to note anything regarding the `tokenAudience`.

I would hope that our tests expose a situation where authentication changes would prevent the integration from functioning as intended. When the tests continue to function, and the developer has reason to believe there will be a problem, the best course of action is another test! 😁

**Review by aiwilliams — APPROVED** [review]

**Pull request merged**